### PR TITLE
feat(ComparisonTable): [NoTicket] Update implementation

### PR DESCRIPTION
**Pull request by kimkwanka**
### What this PR does

This PR merges the current implementation of the `ComparisonTable` with the very similar [CoverageModal](https://github.com/getPopsure/app/blob/8d155d02f5c13bb0fd8a3a59187fd7ee4162a1b3/src/components/CoverageDetails/CoverageModal/CoverageModal.tsx) component used in app which will allow us to replace both eventually. As this is designed as a drop-in-replacement for the ComparisonTable component first and foremost however, the `CoverageModal` will need changing of data structure etc. to be able to utilize this new component.

image

More detailed list of features of this PR:

1. Fix incorrect behaviour (incorrectly aligned elements, non-functional arrow buttons, etc.) in some contexts where reliance on viewport units for calculations fails
2. Add collapsible sections
3. Allow first column and content columns / cells to have sizes other than the previous defaults to better match the new designs. In conjunction with the growContent option this allows for more graceful transitions from desktop to mobile sizes.
4. Add the ability to add a y-offset to the sticky header to be able to use it on website where there's already a sticky nav bar

### Why is this needed?

To fix existing errors in the ComparisonTable while being able to replace the CoverageModal tables with it later down the line.

### Checklist:

- [x] I reviewed my own code
- [x] The changes align with the designs I received  
  Or give a reason why this does not apply:
- [x] I have attached screenshot(s), video(s) or gif(s) showing that the solution is working as expected  
  Or give a reason why this does not apply:
- [x] I have updated the task(s) status on Linear
- [x] All new media is optimized (images, gifs, videos)

### Browser support

My code works in the following browsers:

- [x] Firefox
- [x] Chrome
- [x] Safari
- [x] Edge

### CODE REVIEW

**Review by ruizhen88 — APPROVED** [review]

File: src/lib/components/comparisonTable/components/AccordionItem/AccordionItem.module.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
+@import '../../../../scss/public/colors';
+
+.container {
+  width: var(--growContent, min-content);
+  margin: 0 auto;
+}
+
+.container + .container {
```
Comment (kseha1080) [inline src/lib/components/comparisonTable/components/AccordionItem/AccordionItem.module.scss:8]: Is this a typo?

File: src/lib/components/comparisonTable/style.module.scss [hunk 2]
```diff
@@ -12,6 +12,20 @@
...
 
+.noScrollBars {
+  // Disable the scrollbar in all browsers
+  scrollbar-width: none; /* Firefox */
+  -ms-overflow-style: none; /* Internet Explorer 10+ */
+  &::-webkit-scrollbar {
+    /* WebKit */
+    width: 0;
+    height: 0;
+  }
```
Comment (kseha1080) [inline src/lib/components/comparisonTable/style.module.scss:23]: Don't we need horizontal scroll bars?

Comment (kimkwanka) [reply src/lib/components/comparisonTable/style.module.scss:23]: It's limited to the table content, it does not disable scroll bars globally.

File: src/lib/components/comparisonTable/components/AccordionItem/AccordionItem.module.scss (hunk 1 above)
Comment (kimkwanka) [reply src/lib/components/comparisonTable/components/AccordionItem/AccordionItem.module.scss:8]: No, it's meant to add distance between containers ("if a container follows a container")

Comment (kimkwanka) [reply src/lib/components/comparisonTable/components/AccordionItem/AccordionItem.module.scss:8]: Which means the very first container won't have that margin.

**Review by kseha1080 — APPROVED** [review]

**Pull request merged**